File: 577/577.js
var ans = 0;
var max = 12344
var m = 1;
while(max > 1) {
  for(var n = 1; n < max; n++) {
    ans += (m * (n * (n + 1))) / 2;
  }
  max -= 3;
  m++;
}

console.log(ans)

File: 577/577.py
triNums = [(n * (n + 1)) / 2 for n in range(12344)]
ans = s = sum(triNums)
m = 2

while len(triNums) > 0:
  s -= sum(triNums[-3:])
  triNums = triNums[:-3 or None]
  ans += (s * m)
  m += 1

print ans